src/inputs/ChipTextField.tsx

```diff
@@ -0,0 +1,74 @@
+import { InputHTMLAttributes, KeyboardEvent, useRef, useState } from "react";
+import { useFocus, useTextField } from "react-aria";
+import { usePresentationContext } from "src/components/PresentationContext";
+import { Css } from "src/Css";
+import { maybeCall, useTestIds } from "src/utils";
+
+interface ChipTextFieldProps {
+  // Label is not visible in the component, but required for accessibility purposes
+  label: string;
+  // Current value of the input element
+  value?: string;
+  // Whether the input is required. Adds aria-required attribute
+  required?: boolean;
+  // Sets input's placeholder attribute
+  placeholder?: string;
+  // Whether to call `onBlur` when the user presses the "Escape" key while focused on the input.
+  // @default `true`
+  blurOnEscape?: false;
+  // Callback fired when the user changes the value of the input
+  onChange: (value: string) => void;
+  // Callback fired when focus removes from the input
+  onBlur?: () => void;
+  // Callback fired when focus is set to the input
+  onFocus?: () => void;
+  // Callback when the user presses the "Enter" key while focused on the input
+  onEnter?: () => void;
+}
+
+// A TextField styled to look like a Chip
+export function ChipTextField(props: ChipTextFieldProps) {
+  const { onFocus, onBlur, onEnter, required, label, blurOnEscape = true, ...otherProps } = props;
+  const { fieldProps } = usePresentationContext();
+  const tid = useTestIds(props, "chipInput");
+  const [isFocused, setIsFocused] = useState(false);
+  const { focusProps } = useFocus({
+    onFocus: (e) => {
+      (e.target as HTMLInputElement).select();
+      maybeCall(onFocus);
+    },
+    onBlur: () => maybeCall(onBlur),
+    onFocusChange: setIsFocused,
+  });
+
+  const textFieldProps = {
+    ...otherProps,
+    isRequired: required,
+    "aria-label": label,
+    onKeyDown: (e: KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === "Enter") {
+        maybeCall(onEnter);
+      } else if (blurOnEscape && e.key === "Escape") {
+        (e.target as HTMLInputElement).blur();
+      }
+    },
+  };
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const { inputProps } = useTextField(textFieldProps, inputRef);
+  const typeScale = fieldProps?.typeScale ?? "sm";
+
+  return (
+    <input
+      {...(inputProps as InputHTMLAttributes<HTMLInputElement>)}
+      {...focusProps}
+      ref={inputRef}
+      css={{
+        ...Css[typeScale].dif.aic.br16.pl1.pxPx(10).pyPx(2).gray900.bgGray300.outline0.$,
+        ...(isFocused ? Css.bshFocus.$ : {}),
+      }}
+      {...tid}
+      size={props.value?.length ?? 1}
```

Comment:
**suggestion: (non-blooking):** This size prop looks like it's making some weird behavior. Is there a way to accomplish this with css instead?

<img width="642" alt="Screen Shot 2021-11-04 at 11 58 08 AM" src="https://user-images.githubusercontent.com/16562801/140394072-a765449b-bf7b-4608-87cc-dd8b85d8f263.png">
<img width="642" alt="Screen Shot 2021-11-04 at 11 57 44 AM" src="https://user-images.githubusercontent.com/16562801/140394075-7565132d-95b2-4039-8904-a7ffe8683aac.png">
But I tried some normal words, and it still looked a little off:
<img width="624" alt="Screen Shot 2021-11-04 at 11 55 44 AM" src="https://user-images.githubusercontent.com/16562801/140394083-a3909333-3d88-41b6-90e7-3bbf755b72a3.png">

Reply:
Yea, this is a bummer but there really isn't a great way to do this when using an `<input />` field. It cannot be accomplished using CSS because the `value` is not actually in the DOM, but just an attribute on the input element.  Because it isn't in the DOM then we can't check against the width. 
Potential ways of fixing:
- Use a hidden element that we fill its contents with the save value as the input. Every time that changes, we'd measure the hidden element and use that width to set on the input.
- Use a `<span contenteditable />` rather than an `<input />`. Going this route we'd need to handle things a bit differently... like how we initially select the text when focused would need to use `createRange`. Though, this is a pretty simple input, maybe worth testing out if contenteditable can replace it?

Reply:
@lucyconklin - I just tried this out with contenteditable. Let me know if you want to take a look.